### DISCUSSION TITLE
Implementing mooring lines for WEC device

### DISCUSSION THREAD

**Discussion by Lorenzodecarlo** [DualSPHysics/DualSPHysics#196] [DUALSPHYSICS_V5.2]
Dear members,

I’m a student currently working on a project involving the implementation of moorings for a Wave Energy Converter (WEC) device using DualSPHysics - Moordyn. During my work, I have encountered some doubts and questions for which I would be grateful to receive your valuable help and advice, hoping to do something nice for other users as well.

My questions are as follows:

**1) Implementation of a 3-segment mooring line (chain-fibre rope-chain):**
I am trying to implement a three-segment mooring line, composed of chain-fibre rope-chain, for my WEC device. I would like to know if there is a possibility to do this using DualSPHysics- Moordyn, and if so, if you could provide me with advice on how to proceed or any alternative that could be considered.

**2) Scaling and truncation methods for mooring lines:**
I am interested in understanding the scaling methods used to transition from the "CaseMoorings3D_RealScale_Def" case to the "CaseMoorings3D_Def" case in […\DualSPHysics_v5.2\examples\moordyn\03_WavesMoorings3D]. Specifically, I would like to know if the truncation method was used for mooring lines, and if so, I would like to receive more details on how it was implemented.

**3) Usage of "partbegin" in the batch launch file:**
I would like to understand in more detail how "partbegin" works in the batch launch file. I need to run several simulations with variable maximum durations, extending the duration each time. I would like to know how I can use "partbegin" to manage this situation, but I think that for example ‘-partbegin:100:1 dir’ restarts the simulation from a time instant of 100 seconds and generates the first part 1, while dir indicates the path to the folder where the simulation is located.

I greatly appreciate your time and willingness to answer my questions. 

Thank you in advance for your support and cooperation,

Best regards.

**Comment by imestevez** [comment]
Dear user,

**1)** It is possible to interconnect different moorings lines in this coupling between DualSPHysics-MoorDynPlus by following:
  
  1.1) Define the connection points using `<connects>` section and setting different references for each one:
```
        <moordyn>
                ....
                <connects>
                        <connect conref="1" x="-60.0" y="0.0" z="-20.0" M="1000" V="120"/>
                </connects>
                ....
        </moordyn>
```

  1.2) Define the end nodes of a line:
```
        <moordyn>
                ....
                <lines>
                        <line>  %line 0
                            <vesselconnection bodyref="50" x="-60.0" y="0" z="0" />
                            <connect conref="1" />
                            <length value="20" comment="Unstretched length." units_comment="(m)"/>
                            <segments value="10" />
                        </line>
                        <line>  %line 1
                            <connect conref="1" />
                            <fixconnection x="-100" y="40" z="-50" />
                            <length value="70" comment="Unstretched length." units_comment="(m)"/>
                            <segments value="14" />
                        </line>
                        ...
                </lines>
        </moordyn>
```

**2)** No scaling method has been used for this case; they are different setups in the same folder because they are both 3D cases with waves.

**3)** The restart mode using `-partbegin:begin[:first] dir`, where `begin` specifies the beginning of the simulation starting from a given PART, `dir` indicates where it is located,  and `first` indicates the number of the first PART to be generated. 
However, this option is not yet available to be used within MoorDynPlus, it may be included in the next release.

**Reply by Lorenzodecarlo** [reply]
Thank you very much, your advice was very important. It works and is exactly what I was trying to achieve. I tried extending what you shared with me by 3 segments in the following way, adding properly the second connect (not shown below): 

                        <line>%line 0
                        
                            <vesselconnection bodyref="45" x="-0.8" y="0.0" z="-0.1" />
                            <connect conref="1" />
                            <length value="0.23" comment="Unstretched length." units_comment="(m)" />
                            <segments value="20" />
                        </line>
                        
                        <line>%line 1
                            <connect conref="1" />
                            <connect conref="2" />
                            <length value="0.23" comment="Unstretched length." units_comment="(m)" />
                            <segments value="20" />
                        </line>
                        
                        <line>%line 2
                            <connect conref="2" />
                            <fixconnection x="-1" y="0.0" z="-0.45" />
                            <length value="0.23" comment="Unstretched length." units_comment="(m)" />
                            <segments value="10" />
                        </line>

But when I launch the simulation I get the following error:

Running MoorDyn+ v2.041 (11-01-2023) GPL v3
===================================================

*** Exception (Line::ReadXml) at DualSPHysics/source/Line.cpp:387
Text: Only 2 connections are allowed. Found: 1.


Finished execution (code=1).

Execution aborted.

Where am I going wrong? Is it not possible to extend the use of 2 connectors for a mooring line consisting of 3 segments?  Are there any limitations in this respect? 

I ask because ultimately my aim would be to simulate a mooring system consisting of 4 lines, each consisting of 3 segments. 
Are there any limitations on the number of connectors? 

Thank you in advance for your support and cooperation,

Best regards.

**Comment by imestevez** [comment]
Dear,

Only one `<connect>` is allowed for each `<line>` in the current implementation; it is not related to the number of segments of the mooring line. The use of two connections will be available in the next release of DualSPHysics.

Regards
